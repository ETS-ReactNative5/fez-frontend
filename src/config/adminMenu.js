// menuItems to display
export const menuItems = email => [
    {
        linkTo: '/admin',
        primaryText: 'My dashboard',
        secondaryText: email
    },
    {
        divider: true
    }
];

