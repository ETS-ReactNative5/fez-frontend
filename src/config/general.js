// Authentication
export const SESSION_COOKIE_NAME = 'UQLID';
export const TOKEN_NAME = 'X-Uql-Token';

// URLS
// export const API_URL = process.env.NODE_ENV === 'development' ? 'http://dev-api.library.uq.edu.au:8050/' : 'https://api.library.uq.edu.au/staging/';
export const API_URL = 'https://api.library.uq.edu.au/staging/';
export const AUTH_URL_LOGIN = 'https://auth.library.uq.edu.au/login';
export const AUTH_URL_LOGOUT = 'https://auth.library.uq.edu.au/logout';

// Default values for createNewRecord
export const NEW_RECORD_DEFAULT_VALUES = {
    rek_object_type: 3,
    rek_status: 3,
    fez_record_search_key_ismemberof: [
        {rek_ismemberof: 'UQ:218198'}
    ]
};

export const PUBLICATION_TYPE_GENERIC_DOCUMENT = 202;
export const PUBLICATION_TYPE_AUDIO_DOCUMENT = 263;
export const PUBLICATION_TYPE_BOOK = 174;
export const PUBLICATION_TYPE_BOOK_CHAPTER = 177;
export const PUBLICATION_TYPE_CONFERENCE_PAPER = 130;
export const PUBLICATION_TYPE_CONFERENCE_PROCEEDINGS = 197;
export const PUBLICATION_TYPE_CREATIVE_WORK = 313;
export const PUBLICATION_TYPE_DATA_COLLECTION = 371;
export const PUBLICATION_TYPE_DEPARTMENT_TECHNICAL_REPORT = 181;
export const PUBLICATION_TYPE_DESIGN = 316;
export const PUBLICATION_TYPE_DIGILIB_IMAGE = 228;
export const PUBLICATION_TYPE_IMAGE = 238;
export const PUBLICATION_TYPE_JOURNAL = 294;
export const PUBLICATION_TYPE_JOURNAL_ARTICLE = 179;
export const PUBLICATION_TYPE_MANUSCRIPT = 374;
export const PUBLICATION_TYPE_NEWSPAPER_ARTICLE = 191;
export const PUBLICATION_TYPE_PATENT = 185;
export const PUBLICATION_TYPE_PREPRINT = 204;
export const PUBLICATION_TYPE_RESEARCH_REPORT = 275;
export const PUBLICATION_TYPE_SEMINAR_PAPER = 189;
export const PUBLICATION_TYPE_THESIS = 187;
export const PUBLICATION_TYPE_VIDEO_DOCUMENT = 310;
export const PUBLICATION_TYPE_WORKING_PAPER = 183;

export const publicationTypes = (components) => [
    {
        id: PUBLICATION_TYPE_GENERIC_DOCUMENT,
        name: 'Generic Document',
        class: 'Uqlibrary\\FezCore\\Types\\Generic',
        formComponent: components ? components.GenericDocumentForm : null,
        citationComponent: components ? components.GenericDocumentCitation : null,
        hasFormComponent: true
    },
    {
        id: PUBLICATION_TYPE_AUDIO_DOCUMENT,
        name: 'Audio Document',
        class: 'Uqlibrary\\FezCore\\Types\\Audio',
        formComponent: components ? components.AudioDocumentForm : null,
        citationComponent: components ? components.AudioDocumentCitation : null,
        hasFormComponent: true
    },
    {
        id: PUBLICATION_TYPE_BOOK,
        name: 'Book',
        class: 'Uqlibrary\\FezCore\\Types\\Book',
        subtypeVocabId: 453581,
        isFavourite: true,
        formComponent: components ? components.BookForm : null,
        citationComponent: components ? components.BookCitation : null,
        hasFormComponent: true
    },
    {
        id: PUBLICATION_TYPE_BOOK_CHAPTER,
        name: 'Book Chapter',
        class: 'Uqlibrary\\FezCore\\Types\\BookChapter',
        subtypeVocabId: 453588,
        isFavourite: true,
        formComponent: components ? components.BookChapterForm : null,
        citationComponent: components ? components.BookChapterCitation : null,
        hasFormComponent: true
    },
    {
        id: PUBLICATION_TYPE_CONFERENCE_PAPER,
        name: 'Conference Paper',
        class: 'Uqlibrary\\FezCore\\Types\\ConferencePaper',
        subtypeVocabId: 453596,
        isFavourite: true,
        formComponent: components ? components.ConferencePaperForm : null,
        citationComponent: components ? components.ConferencePaperCitation : null,
        hasFormComponent: true
    },
    {
        id: PUBLICATION_TYPE_CONFERENCE_PROCEEDINGS,
        name: 'Conference Proceedings',
        class: 'Uqlibrary\\FezCore\\Types\\ConferenceProceedings',
        hasFormComponent: true
    },
    {
        id: PUBLICATION_TYPE_CREATIVE_WORK,
        name: 'Creative Work',
        class: 'Uqlibrary\\FezCore\\Types\\CreativeWork',
        citationComponent: components ? components.CreativeWorkCitation : null,
        formComponent: components ? components.CreativeWorkForm : null,
        vocabId: 453594,
        hasFormComponent: true
    },
    {
        id: PUBLICATION_TYPE_DATA_COLLECTION,
        name: 'Data Collection',
        class: 'Uqlibrary\\FezCore\\Types\\DataCollection',
        hasFormComponent: false
    },
    {
        id: PUBLICATION_TYPE_DEPARTMENT_TECHNICAL_REPORT,
        name: 'Department Technical Report',
        class: 'Uqlibrary\\FezCore\\Types\\DepartmentTechnicalReport',
        hasFormComponent: true
    },
    {
        id: PUBLICATION_TYPE_DESIGN,
        name: 'Design',
        class: 'Uqlibrary\\FezCore\\Types\\Design',
        hasFormComponent: true
    },
    {
        id: PUBLICATION_TYPE_DIGILIB_IMAGE,
        name: 'Digilib Image',
        class: 'Uqlibrary\\FezCore\\Types\\DigilibImage',
        hasFormComponent: false
    },
    {
        id: PUBLICATION_TYPE_IMAGE,
        name: 'Image',
        class: 'Uqlibrary\\FezCore\\Types\\Image',
        hasFormComponent: true
    },
    {
        id: PUBLICATION_TYPE_JOURNAL,
        name: 'Journal',
        class: 'Uqlibrary\\FezCore\\Types\\Journal',
        hasFormComponent: false
    },
    {
        id: PUBLICATION_TYPE_JOURNAL_ARTICLE,
        name: 'Journal Article',
        class: 'Uqlibrary\\FezCore\\Types\\JournalArticle',
        subtypeVocabId: 453573,
        isFavourite: true,
        formComponent: components ? components.JournalArticleForm : null,
        citationComponent: components ? components.JournalArticleCitation : null,
        hasFormComponent: true
    },
    {
        id: PUBLICATION_TYPE_MANUSCRIPT,
        name: 'Manuscript',
        class: 'Uqlibrary\\FezCore\\Types\\Manuscript',
        citationComponent: components ? components.ManuscriptCitation : null,
        hasFormComponent: false
    },
    {
        id: PUBLICATION_TYPE_NEWSPAPER_ARTICLE,
        name: 'Newspaper Article',
        class: 'Uqlibrary\\FezCore\\Types\\NewspaperArticle',
        hasFormComponent: true
    },
    {
        id: PUBLICATION_TYPE_PATENT,
        name: 'Patent',
        class: 'Uqlibrary\\FezCore\\Types\\Patent',
        formComponent: components ? components.PatentForm : null,
        hasFormComponent: true
    },
    {
        id: PUBLICATION_TYPE_PREPRINT,
        name: 'Preprint',
        class: 'Uqlibrary\\FezCore\\Types\\Preprint',
        formComponent: components ? components.PreprintForm : null,
        citationComponent: components ? components.PreprintCitation : null,
        hasFormComponent: true
    },
    {
        id: PUBLICATION_TYPE_RESEARCH_REPORT,
        name: 'Research Report',
        class: 'Uqlibrary\\FezCore\\Types\\ResearchReport',
        formComponent: components ? components.ResearchReportForm : null,
        citationComponent: components ? components.ResearchReportCitation : null,
        hasFormComponent: true
    },
    {
        id: PUBLICATION_TYPE_SEMINAR_PAPER,
        name: 'Seminar Paper',
        class: 'Uqlibrary\\FezCore\\Types\\SeminarPaper',
        formComponent: components ? components.SeminarPaperForm : null,
        citationComponent: components ? components.SeminarPaperCitation : null,
        hasFormComponent: true
    },
    {
        id: PUBLICATION_TYPE_THESIS,
        name: 'Thesis',
        class: 'Uqlibrary\\FezCore\\Types\\Thesis',
        hasFormComponent: true
    },
    {
        id: PUBLICATION_TYPE_VIDEO_DOCUMENT,
        name: 'Video Document',
        class: 'Uqlibrary\\FezCore\\Types\\Video',
        formComponent: components ? components.VideoDocumentForm : null,
        hasFormComponent: true
    },
    {
        id: PUBLICATION_TYPE_WORKING_PAPER,
        name: 'Working Paper',
        class: 'Uqlibrary\\FezCore\\Types\\WorkingPaper',
        formComponent: components ? components.WorkingPaperForm : null,
        hasFormComponent: true
    }
];

export const openAccessIdLookup = {
    453693: 'DOI',
    453694: 'Link (no DOI)',
    453695: 'File (Publisher version)',
    453696: 'File (Author post-print)',
    453697: 'Other',
};

export const OrgUnitsVocabId = 453703;

