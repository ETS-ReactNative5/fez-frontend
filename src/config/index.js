export {
    API_URL,
    APP_URL,
    AUTH_URL_LOGIN,
    AUTH_URL_LOGOUT,
    SESSION_COOKIE_NAME,
    TOKEN_NAME,
    ORCID_BASE_URL,
    ORCID_CLIENT_ID,
    ORCID_AUTHORIZATION_URL,
    FACET_TYPE_FILTER,
    FACET_TYPE_RANGE,
    FEZ_URL
} from './general';

export {api, generateCancelToken, cache} from './axios';
export {default as theme} from './theme';
export * as validation from './validation';
export * as general from './general';
export {publicationTypes} from './general';
export * as routes from './routes';
