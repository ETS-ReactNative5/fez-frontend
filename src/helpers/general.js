export const leftJoin = (objArr1, objArr2, key1, key2) => {
    if (!objArr2) {
        return objArr1;
    }
    return objArr1.map(anObj1 => ({
        ...objArr2.find(anObj2 => anObj1[key1] === anObj2[key2]),
        ...anObj1,
    }));
};

export const stripHtml = html => {
    const temporalDivElement = document.createElement('div');
    temporalDivElement.innerHTML = html;
    return temporalDivElement.textContent || temporalDivElement.innerText || '';
};

export const doesListContainItem = (list, term) => {
    return list.some(sort => (typeof sort === 'object' ? sort.value === term : sort === term));
};
