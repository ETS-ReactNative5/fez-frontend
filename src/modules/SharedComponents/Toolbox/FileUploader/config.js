export const CLOSED_ACCESS_ID = 8;
export const OPEN_ACCESS_ID = 9;
export const SIZE_BASE = 1000;

export const DEFAULT_FILE_UPLOAD_LIMIT = 10;
export const DEFAULT_MAX_FILE_SIZE = 8;

export const SIZE_UNIT_B = 'B';
export const SIZE_UNIT_K = 'K';
export const SIZE_UNIT_M = 'M';
export const SIZE_UNIT_G = 'G';

export const FILE_META_KEY_ACCESS_CONDITION = 'access_condition_id';
export const FILE_META_KEY_EMBARGO_DATE = 'date';

export const SIZE_UNITS = [SIZE_UNIT_B, SIZE_UNIT_K, SIZE_UNIT_M, SIZE_UNIT_G];

export const FILE_NAME_RESTRICTION = /^(?=^\S*$)(?=^[a-z\d\-_]+(\.\d{3}|\.r\d{2,3}|\.part\d{1,3})?\.[^\.]+$)(?=.{1,45}$)(?!(web_|preview_|thumbnail_|stream_|fezacml_|presmd_|\d))[a-z\d\-_\.]+/;

/**
 * Note: ext -> mimetype is a 1-n relation, but we only care about one value
 */
export const MIME_TYPE_WHITELIST = {
    pdf: 'application/pdf',
    jpe: 'image/jpeg',
    jpeg: 'image/jpeg',
    jpg: 'image/jpeg',
    gif: 'image/gif',
    tif: 'image/tiff',
    tiff: 'image/tiff',
    png: 'image/png',
    zip: 'application/zip',
    tar: 'application/x-tar',
    gz: 'application/gzip',
    '7z': 'application/x-7z-compressed',
    rar: 'application/x-rar-compressed',
    mk3d: 'video/x-matroska',
    mks: 'video/x-matroska',
    mkv: 'video/x-matroska',
    qt: 'video/quicktime',
    mov: 'video/quicktime',
    mp4: 'video/mp4',
    mp4v: 'video/mp4',
    mp3: 'audio/mp3',
    wav: 'audio/wav',
    mpe: 'video/mpeg',
    mpeg: 'video/mpeg',
    mpg: 'video/mpeg',
    m1v: 'video/mpeg',
    m2v: 'video/mpeg',
    wmv: 'video/x-ms-wmv',
    wma: 'audio/x-ms-wma',
    avi: 'video/x-msvideo',
    m4a: 'audio/x-m4a',
    csv: 'text/csv',
    xla: 'application/vnd.ms-excel',
    xlc: 'application/vnd.ms-excel',
    xlm: 'application/vnd.ms-excel',
    xls: 'application/vnd.ms-excel',
    xlt: 'application/vnd.ms-excel',
    xlw: 'application/vnd.ms-excel',
    xlsx: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
    ods: 'application/vnd.oasis.opendocument.spreadsheet',
    gsheet: 'application/vnd.google-apps.spreadsheet',
};
