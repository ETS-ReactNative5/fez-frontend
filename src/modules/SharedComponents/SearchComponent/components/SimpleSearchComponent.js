import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import IconButton from '@material-ui/core/IconButton';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import Tooltip from '@material-ui/core/Tooltip';

import Search from '@material-ui/icons/Search';
import ArrowBack from '@material-ui/icons/ArrowBack';

import {MAX_PUBLIC_SEARCH_TEXT_LENGTH} from 'config/general';
import {locale} from 'locale';

import {withStyles} from '@material-ui/core/styles';
import Hidden from '@material-ui/core/Hidden';
import Fade from '@material-ui/core/Fade';

const styles = theme => ({
    searchIconPrefix: {
        fill: theme.palette.secondary.main,
        opacity: 0.66
    },
    inHeader: {
        backgroundColor: theme.palette.white.main,
        '& input[type="search"]::-webkit-search-cancel-button': {
            display: 'none'
        }
    },
    searchIconMobile: {
        fill: theme.palette.white.main
    },
    mobileBackArrow: {
        height: 50,
        width: 50,
        fill: theme.palette.secondary.main,
        opacity: 0.5
    },
    mobileBackArrowButton: {
        height: 70,
        width: 70,
    },
    mobileHeader: {
        backgroundColor: theme.palette.white.main,
        position: 'absolute',
        top: 0,
        left: 0,
        width: '100%',
        height: 70
    },
    mobileSearchInput: {
        height: 70,
        margin: ' 0 12px',
        '& input': {
            marginTop: 10,
            fontSize: 32,
            lineHeight: 70,
            fontWeight: theme.typography.fontWeightNormal
        }
    }
});

export class SimpleSearchComponent extends PureComponent {
    static propTypes = {
        className: PropTypes.string,

        searchText: PropTypes.string,

        isInHeader: PropTypes.bool,
        showSearchButton: PropTypes.bool,
        showMobileSearchButton: PropTypes.bool,
        showAdvancedSearchButton: PropTypes.bool,
        showPrefixIcon: PropTypes.bool,

        onSearch: PropTypes.func,
        onSearchTextChange: PropTypes.func.isRequired,
        onToggleSearchMode: PropTypes.func,
        onInvalidSearch: PropTypes.func,

        classes: PropTypes.object
    };
    static defaultProps = {
        searchText: '',

        isInHeader: false,
        showSearchButton: false,
        showMobileSearchButton: false,
        showAdvancedSearchButton: false,
        showPrefixIcon: false,

        onSearch: () => {
        },
        onToggleSearchMode: () => {
        },
        onInvalidSearch: () => {
        }
    };

    constructor(props) {
        super(props);
        this.state = {
            showMobile: false
        };
    }

    searchTextValidationMessage = (value) => {
        if (value.trim().length > MAX_PUBLIC_SEARCH_TEXT_LENGTH) {
            return locale.validationErrors.maxLength.replace('[max]', MAX_PUBLIC_SEARCH_TEXT_LENGTH);
        }

        return null;
    };

    _handleToggleMobile = () => {
        this.setState({
            ...this.state,
            showMobile: !this.state.showMobile
        }, () => {
            if (this.state.showMobile) {
                document.getElementById('mobileSearchField') && document.getElementById('mobileSearchField').focus();
            }
        });
    };

    _handleSearchTextChange = (event) => {
        this.props.onSearchTextChange(event.target.value);
    };

    _handleSearchMode = () => {
        if (!!this.props.onToggleSearchMode) {
            this.props.onToggleSearchMode();
        }
    };

    _handleSearch = (event) => {
        if (event && event.key && (event.key !== 'Enter')) return;

        if (this.props.searchText.trim().length > MAX_PUBLIC_SEARCH_TEXT_LENGTH) {
            this.props.onInvalidSearch(locale.validationErrors.maxLength.replace('[max]', MAX_PUBLIC_SEARCH_TEXT_LENGTH));
            return;
        }

        // Hide the mobile search bar after performing a search
        this.setState({
            ...this.state,
            showMobile: false
        });

        // Perform search
        this.props.onSearch();

        // Blur the input so the mobile keyboard is deactivated
        event && event.target && event.target.blur();
    };

    render() {
        const txt = locale.components.searchComponent;
        const {classes} = this.props;
        return (
            <React.Fragment>
                {
                    this.props.isInHeader ?
                        <React.Fragment>
                            {/* DESKTOP in header */}
                            <Hidden xsDown>
                                <Grid container spacing={16} alignItems={'center'} alignContent={'center'} justify={'center'} className={classes.inHeader}>
                                    {
                                        this.props.showPrefixIcon &&
                                        <Grid item>
                                            <Search className={classes.searchIconPrefix}/>
                                        </Grid>
                                    }
                                    <Grid item style={{flexGrow: 1}}>
                                        <TextField
                                            type="search"
                                            fullWidth
                                            label={!this.props.isInHeader && txt.searchBoxPlaceholder}
                                            placeholder={this.props.isInHeader ? txt.searchBoxPlaceholder : txt.searchBoxHint}
                                            aria-label={txt.ariaInputLabel}
                                            onChange={this._handleSearchTextChange}
                                            onKeyPress={this._handleSearch}
                                            value={this.props.searchText}
                                            InputProps={{disableUnderline: true}}
                                            error={this.searchTextValidationMessage(this.props.searchText)}/>
                                    </Grid>
                                </Grid>
                            </Hidden>
                            {/* MOBILE in header */}
                            <Hidden smUp>
                                {
                                    !this.state.showMobile ?
                                        <Tooltip title={txt.searchBoxPlaceholder} placement="bottom-end" TransitionComponent={Fade} TransitionProps={{ timeout: 300 }}>
                                            <IconButton
                                                onClick={this._handleToggleMobile}
                                                aria-label={txt.mobileSearchButtonAriaLabel}>
                                                <Search className={classes.searchIconMobile}/>
                                            </IconButton>
                                        </Tooltip>
                                        :
                                        <div className={classes.mobileHeader}>
                                            <Grid container spacing={0} alignItems={'stretch'} justify={'center'}>
                                                {
                                                    this.props.showMobileSearchButton && this.state.showMobile &&
                                                    <Hidden smUp>
                                                        <Grid item>
                                                            <Button onClick={this._handleToggleMobile} className={classes.mobileBackArrowButton}>
                                                                <ArrowBack className={classes.mobileBackArrow}/>
                                                            </Button>
                                                        </Grid>
                                                    </Hidden>
                                                }
                                                <Grid item style={{flexGrow: 1}}>
                                                    <TextField
                                                        className={classes.mobileSearchInput}
                                                        type="search"
                                                        id="mobileSearchField"
                                                        fullWidth
                                                        label={!this.props.isInHeader && txt.searchBoxPlaceholder}
                                                        placeholder={this.props.isInHeader ? txt.searchBoxPlaceholder : txt.searchBoxHint}
                                                        aria-label={txt.ariaInputLabel}
                                                        onChange={this._handleSearchTextChange}
                                                        onKeyPress={this._handleSearch}
                                                        value={this.props.searchText}
                                                        InputProps={{disableUnderline: true}}
                                                        error={this.searchTextValidationMessage(this.props.searchText)}/>
                                                </Grid>
                                            </Grid>
                                        </div>
                                }
                            </Hidden>
                        </React.Fragment>
                        :
                        <React.Fragment>
                            {/* NOT in header */}
                            <Grid container spacing={16} alignContent={'flex-end'} alignItems={'flex-end'}>
                                <Grid item style={{flexGrow: 1}}>
                                    <TextField
                                        type="search"
                                        fullWidth
                                        label={!this.props.isInHeader && txt.searchBoxPlaceholder}
                                        placeholder={this.props.isInHeader ? txt.searchBoxPlaceholder : txt.searchBoxHint}
                                        aria-label={txt.ariaInputLabel}
                                        onChange={this._handleSearchTextChange}
                                        onKeyPress={this._handleSearch}
                                        value={this.props.searchText}
                                        error={this.searchTextValidationMessage(this.props.searchText)}
                                    />
                                </Grid>
                                <Grid item xs={12} sm={'auto'}>
                                    <Button
                                        children={txt.searchButtonText}
                                        aria-label={txt.searchButtonAriaLabel}
                                        variant={'raised'}
                                        color={'primary'}
                                        disabled={!!this.searchTextValidationMessage(this.props.searchText)}
                                        onClick={this._handleSearch}
                                        fullWidth
                                    />
                                </Grid>
                                <Grid item xs={12} sm={'auto'}>
                                    <Button
                                        variant={'raised'}
                                        children={txt.advancedSearchButtonText}
                                        aria-label={txt.advancedSearchButtonAriaLabel}
                                        onClick={this._handleSearchMode}
                                        className="advancedButton"
                                        fullWidth
                                    />
                                </Grid>
                            </Grid>
                        </React.Fragment>
                }
            </React.Fragment>
        );
    }
}

export default withStyles(styles, {withTheme: true})(SimpleSearchComponent);
