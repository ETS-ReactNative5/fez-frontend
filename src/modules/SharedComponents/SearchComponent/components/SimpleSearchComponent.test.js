import SimpleSearchComponent from './SimpleSearchComponent';
import * as constants from 'config/general';

function setup(testProps, isShallow = true){
    const props = {
        searchText: '',
        className: 'simple-search',

        showSearchButton: false,
        showMobileSearchButton: false,
        showAdvancedSearchButton: false,
        showPrefixIcon: false,

        isInHeader: false,

        onSearch: jest.fn(),
        onInvalidSearch: jest.fn(),
        onToggle: jest.fn(),
        onSearchTextChange: jest.fn(),

        ...testProps
    };

    return getElement(SimpleSearchComponent, props, isShallow);
}

describe('SimpleSearchComponent', () => {
    it('should render default view', () => {
        const wrapper = setup({});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render show mobile button', () => {
        const wrapper = setup({showMobileSearchButton: true});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render show prefix icon in the search box', () => {
        const wrapper = setup({showPrefixIcon: true});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with a class "header" for use in AppBar', () => {
        const wrapper = setup({isInHeader: true});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should set search value from prop', () => {
        const wrapper = setup({showAdvancedSearchButton: true, searchText: 'i feel lucky'});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should update search text field', () => {
        const testFn = jest.fn();
        const wrapper = setup({onSearchTextChange: testFn});

        wrapper.instance().handleSearchTextChange({}, 'new search value');

        expect(testFn).toHaveBeenCalledWith('new search value');
    });

    it('should not submit search if ENTER wasn\'t pressed', () => {
        const testMethod = jest.fn();
        const wrapper = setup({onSearch: testMethod});

        wrapper.instance().handleSearch({key: 'a'});
        wrapper.update();

        expect(testMethod).not.toHaveBeenCalled();
    });

    it('should submit search if search text is not null and ENTER is pressed', () => {
        const testMethod = jest.fn();
        const wrapper = setup({searchText: 'i feel lucky', onSearch: testMethod});

        wrapper.instance().handleSearch({key: 'Enter'});
        wrapper.update();

        expect(testMethod).toHaveBeenCalled();
    });

    it('should toggle search mode', () => {
        const testToggleFn = jest.fn();
        const wrapper = setup({showAdvancedSearchButton: true, onToggleSearchMode: testToggleFn});
        wrapper.instance().handleSearchMode();
        expect(testToggleFn).toHaveBeenCalled();
    });

    it('should toggle mobile search', () => {
        const wrapper = setup({});
        wrapper.instance().handleToggleMobile();
        expect(wrapper.state().showMobile).toBe(true);
    });

    it('should handle search and notify with error message for max length for search text', () => {
        const testOnInvalidSearchFn = jest.fn();
        const wrapper = setup({searchText: 'this is way too long', onInvalidSearch: testOnInvalidSearchFn});

        constants.MAX_PUBLIC_SEARCH_TEXT_LENGTH = 5;

        wrapper.instance().handleSearch();
        expect(testOnInvalidSearchFn).toHaveBeenCalledWith('Must be 5 characters or less');
    });

    it('_searchTextValidationMessage() should return a message for being too long', () => {
        const wrapper = setup({});
        wrapper.setState({searchText: 'this is way too long'});
        constants.MAX_PUBLIC_SEARCH_TEXT_LENGTH = 5;
        wrapper.update();
        expect(wrapper.instance()._searchTextValidationMessage(wrapper.state().searchText)).toEqual('Must be 5 characters or less');
    });

    it('_searchTextValidationMessage() should return false for being fine', () => {
        const wrapper = setup({});
        constants.MAX_PUBLIC_SEARCH_TEXT_LENGTH = 20;
        wrapper.setState({searchText: 'this is fine'});
        wrapper.update();
        expect(wrapper.instance()._searchTextValidationMessage(wrapper.state().searchText)).toEqual(null);
    });
});
