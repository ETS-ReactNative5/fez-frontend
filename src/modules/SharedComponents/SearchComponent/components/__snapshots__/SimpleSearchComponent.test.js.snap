// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleSearchComponent should render default view 1`] = `
<div
  className="searchComponent false simple-search"
>
  <div
    className="columns is-gapless"
  >
    <div
      className="column search-field is-gapless hideMobile"
    >
      <div
        className="columns is-gapless search-field is-mobile"
      >
        <div
          className="column"
        >
          <TextField
            InputProps={null}
            aria-label="Enter your search query to search eSpace and then press Enter"
            error={null}
            fullWidth={true}
            id="searchField"
            label="Search eSpace"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Enter your search query to search all fields in eSpace"
            required={false}
            select={false}
            style={Object {}}
            type="search"
            value=""
          />
        </div>
        <div
          className="is-hidden-tablet mobileSpacer"
        />
      </div>
    </div>
    <div
      className="column is-narrow searchButtonWrapper"
    >
      <WithStyles(Button)
        aria-label="Click to search eSpace"
        color="primary"
        disabled={false}
        fullWidth={true}
        onClick={[Function]}
        variant="raised"
      >
        Search
      </WithStyles(Button)>
    </div>
  </div>
</div>
`;

exports[`SimpleSearchComponent should render show mobile button 1`] = `
<div
  className="searchComponent false simple-search"
>
  <div
    className="columns is-gapless"
  >
    <div
      className="column search-field is-gapless hideMobile"
    >
      <div
        className="columns is-gapless search-field is-mobile"
      >
        <div
          className="column is-narrow searchIconPrefix is-hidden-tablet"
        >
          <IconButton
            className="mobileBackArrow"
            disableTouchRipple={false}
            disabled={false}
            iconStyle={Object {}}
            onClick={[Function]}
            tooltipPosition="bottom-center"
            touch={false}
          >
            <NavigationArrowBack />
          </IconButton>
        </div>
        <div
          className="column"
        >
          <TextField
            InputProps={null}
            aria-label="Enter your search query to search eSpace and then press Enter"
            error={null}
            fullWidth={true}
            id="searchField"
            label="Search eSpace"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Enter your search query to search all fields in eSpace"
            required={false}
            select={false}
            style={Object {}}
            type="search"
            value=""
          />
        </div>
        <div
          className="is-hidden-tablet mobileSpacer"
        />
      </div>
    </div>
    <div
      className="column is-narrow is-hidden-tablet"
    >
      <IconButton
        aria-label="Click to search eSpace"
        className="searchButton"
        disableTouchRipple={false}
        disabled={false}
        hoveredStyle={
          Object {
            "backgroundColor": "rgba(0,0,0,0.3)",
            "borderRadius": "50%",
          }
        }
        iconStyle={Object {}}
        onClick={[Function]}
        tooltipPosition="bottom-center"
        touch={false}
      >
        <ActionSearch />
      </IconButton>
    </div>
    <div
      className="column is-narrow searchButtonWrapper"
    >
      <WithStyles(Button)
        aria-label="Click to search eSpace"
        color="primary"
        disabled={false}
        fullWidth={true}
        onClick={[Function]}
        variant="raised"
      >
        Search
      </WithStyles(Button)>
    </div>
  </div>
</div>
`;

exports[`SimpleSearchComponent should render show prefix icon in the search box 1`] = `
<div
  className="searchComponent false simple-search"
>
  <div
    className="columns is-gapless"
  >
    <div
      className="column search-field is-gapless hideMobile"
    >
      <div
        className="columns is-gapless search-field is-mobile"
      >
        <div
          className="column is-narrow searchIconPrefix is-hidden-mobile"
        >
          <ActionSearch />
        </div>
        <div
          className="column"
        >
          <TextField
            InputProps={null}
            aria-label="Enter your search query to search eSpace and then press Enter"
            error={null}
            fullWidth={true}
            id="searchField"
            label="Search eSpace"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Enter your search query to search all fields in eSpace"
            required={false}
            select={false}
            style={Object {}}
            type="search"
            value=""
          />
        </div>
        <div
          className="is-hidden-tablet mobileSpacer"
        />
      </div>
    </div>
    <div
      className="column is-narrow searchButtonWrapper"
    >
      <WithStyles(Button)
        aria-label="Click to search eSpace"
        color="primary"
        disabled={false}
        fullWidth={true}
        onClick={[Function]}
        variant="raised"
      >
        Search
      </WithStyles(Button)>
    </div>
  </div>
</div>
`;

exports[`SimpleSearchComponent should render with a class "header" for use in AppBar 1`] = `
<div
  className="searchComponent header simple-search"
>
  <div
    className="columns is-gapless"
  >
    <div
      className="column search-field is-gapless hideMobile"
    >
      <div
        className="columns is-gapless search-field is-mobile"
      >
        <div
          className="column"
        >
          <TextField
            InputProps={
              Object {
                "disableUnderline": true,
              }
            }
            aria-label="Enter your search query to search eSpace and then press Enter"
            error={null}
            fullWidth={true}
            id="searchField"
            label={false}
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Search eSpace"
            required={false}
            select={false}
            style={
              Object {
                "padding": 8,
              }
            }
            type="search"
            value=""
          />
        </div>
        <div
          className="is-hidden-tablet mobileSpacer"
        />
      </div>
    </div>
    <div
      className="column is-narrow searchButtonWrapper"
    >
      <WithStyles(Button)
        aria-label="Click to search eSpace"
        color="primary"
        disabled={false}
        fullWidth={true}
        onClick={[Function]}
        variant="raised"
      >
        Search
      </WithStyles(Button)>
    </div>
  </div>
</div>
`;

exports[`SimpleSearchComponent should set search value from prop 1`] = `
<div
  className="searchComponent false simple-search"
>
  <div
    className="columns is-gapless"
  >
    <div
      className="column search-field is-gapless hideMobile"
    >
      <div
        className="columns is-gapless search-field is-mobile"
      >
        <div
          className="column"
        >
          <TextField
            InputProps={null}
            aria-label="Enter your search query to search eSpace and then press Enter"
            error={null}
            fullWidth={true}
            id="searchField"
            label="Search eSpace"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Enter your search query to search all fields in eSpace"
            required={false}
            select={false}
            style={Object {}}
            type="search"
            value="i feel lucky"
          />
        </div>
        <div
          className="is-hidden-tablet mobileSpacer"
        />
      </div>
    </div>
    <div
      className="column is-narrow searchButtonWrapper"
    >
      <WithStyles(Button)
        aria-label="Click to search eSpace"
        color="primary"
        disabled={false}
        fullWidth={true}
        onClick={[Function]}
        variant="raised"
      >
        Search
      </WithStyles(Button)>
    </div>
    <div
      className="column is-narrow advancedSearchButtonWrapper"
    >
      <WithStyles(Button)
        aria-label="Click to switch to Advanced search"
        className="advancedButton"
        fullWidth={true}
        onClick={[Function]}
        variant="raised"
      >
        Advanced search
      </WithStyles(Button)>
    </div>
  </div>
</div>
`;
